uniffi/src/cli/swift.rs

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use anyhow::Result;
+use camino::Utf8PathBuf;
+use clap::{Args, Parser};
+
+use uniffi_bindgen::bindings::{generate_swift_bindings, SwiftBindingsOptions};
+
+#[derive(Debug, Parser)]
+#[command(version, about, long_about = None)]
+struct Cli {
+    #[command(flatten)]
+    kinds: Kinds,
+    /// Library path to generate bindings for
+    library_path: Utf8PathBuf,
+    /// Directory to generate files in
+    out_dir: Utf8PathBuf,
+    /// Generate a XCFramework-compatible modulemap
+    #[arg(long)]
+    xcframework: bool,
+    /// module name for the generated modulemap
+    #[arg(long)]
+    module_name: Option<String>,
+    /// filename for the generate modulemap
+    #[arg(long)]
+    modulemap_filename: Option<String>,
+    /// Whether we should exclude dependencies when running "cargo metadata".
+    /// This will mean external types may not be resolved if they are implemented in crates
+    /// outside of this workspace.
+    /// This can be used in environments when all types are in the namespace and fetching
+    /// all sub-dependencies causes obscure platform specific problems.
+    #[clap(long)]
+    metadata_no_deps: bool,
+}
+
+#[derive(Debug, Args)]
+#[group(required = true, multiple = true)]
+struct Kinds {
```

Comment:
Would it be helpful to have a `default`? Like "building `.Xcframework`" could be the default. So we can set the `bool` values accordingly.

Reply:
A default sounds good, but I'm not sure what default you're suggesting.  Is it `--xcframework` is the default and `--no-xcframework` would disable it?  Or is it that generating one these kinds of sources is the default?

Reply:
I would say we could add the default to always be `xcframework`, which I think will be most of the time the case. We need Swift specific settings at the moment just for the documentation. So we could shorten the command when building for `xcframework` and don't have to pass any parameters.

Reply:
After thinking about this a bit more, I'm not so sure.  `--no-xcframework` seems a bit awkward as a flag.  It's harder to describe since it's a negative, and it also sort-of implies that if you don't have it then you're building an xcframework somehow.  Something like `--not-for-xcframework` is more precise and easier to describe, but much harder to type out.

How about we leave it as-is for now, we can always change the flag later.